parsers/saarland.py

```diff
@@ -0,0 +1,199 @@
+    global base_data
+
+    components_all = []
+
+    notices_list = get_notices(notices)
+
+    for component in components:
+        component_string = 'dazu ' + component['name']
+        if len(component['notices']) > 0:
+            component_notices_list = get_notices(component['notices'], True)
+            component_string += ' (%s)' % ', '.join(component_notices_list)
+        components_all.append(component_string)
+
+    return components_all + notices_list
+
+
+def build_hours(opening_hours):
+    if not opening_hours:
+        return []
+
+    start = datetime.datetime.strptime(
+        opening_hours['start'], UTC_DATE_STRING).time()
+    end = datetime.datetime.strptime(
+        opening_hours['end'], UTC_DATE_STRING).time()
+    return [
+        'verfügbar von %s bis %s Uhr' %
+        (datetime.time.strftime(start, "%H:%M"),
+            datetime.time.strftime(end, "%H:%M"))
+    ]
+
+
+def build_location(description):
+    return ['Ort: %s' % description] if description else []
+
+
+def parse_url(url, today=False):
+    global base_data
+
+    canteen = LazyBuilder()
+    with urlopen(url) as response:
+        data = json.loads(response.read().decode())
+
+    for day in data['days']:
+        date = day.get('date')
+
+        if today and (datetime.datetime.now().date() !=
+                      datetime.datetime.strptime(date, UTC_DATE_STRING).date()):
+            continue
+
+        for counter in day['counters']:
+            counter_name = counter['displayName']
+            counter_description = counter['description']
+            counter_hours = counter.get('openingHours')
+            counter_color = counter.get('color')
+
+            for meal in counter['meals']:
+                if 'knownMealId' in meal:
+                    print('knownMealId: %s' % meal['knownMealId'])
+
+                meal_name = meal['name']
+                if 'category' in meal:
+                    meal_name = '%s: %s' % (meal['category'], meal_name)
+
+                meal_notes = (
+                    build_location(counter_description) +
+                    build_hours(counter_hours) +
+                    build_notes(meal['notices'], meal['components']))
+
+                meal_prices = {}
+                if 'prices' in meal:
+                    prices = meal['prices']
+                    for role in prices:
+                        if role in base_data['roles']:
+                            meal_prices[base_data['roles'][role]] = prices[role]
+                        else:
+                            print('Ignoring price for unknown role %s.' % role)
+
+                if 'pricingNotice' in meal:
+                    meal_notes.append(meal['pricingNotice'])
+
+                canteen.addMeal(date, counter_name,
+                                meal_name, meal_notes, meal_prices)
+
+    return canteen.toXMLFeed()
+
+
+with urlopen(URL_BASE + URL_BASE_DATA) as response:
+    data = json.loads(response.read().decode())
```

Comment:
Requesting this in the main scope is not a good idea. It leads to executing network requests when simply including the module, which happens even if the parser is not used.
Also depending on how this is hosted, it might be that this is executed once when starting the application and then never updated again.

So I suggest simply moving this into a separate function that gets called in `parse_url` and define the parsers manually (they currently (see #78) have to be defined manually in openmesa.org anyway and they probably don't change that often). This way you can also avoid the `global` definitions.

Reply:
Humm, okay, so you suggest hardcoding the different locations once so that they cannot be updated automatically when the `BaseData` has been updated? Or am I missing some way this would be possible without loading the `BaseData` in the main scope?

Reply:
> […] hardcoding the different locations once […]

Yes, I think this is the best solution for now.

> Or am I missing some way this would be possible without loading the BaseData in the main scope?

I don't think this is possible right now, but we can certainly discuss adding a feature that allows fetching the canteens dynamically, maybe together with support for actually creating the new canteens at openmensa.org automatically (through extended metadata).

Reply:
Okay, I've changed it now.